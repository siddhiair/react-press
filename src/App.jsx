import "./App.css";
import Homepage from "./pages/Homepage";
import Header from "./components/Header";

function App() {
  return (
    <>
      <Header />
      <Homepage />
    </>
  );
}

export default App;
