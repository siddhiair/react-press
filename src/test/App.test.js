/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import App from "../App";

describe("<App />", () => {
  test("Should render homepage", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
