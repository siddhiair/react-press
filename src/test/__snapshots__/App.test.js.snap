// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render homepage 1`] = `
<DocumentFragment>
  <div>
    <header
      class="bg-orange-100 py-2"
    >
      <div
        class="container"
      >
        <div
          class="flex gap-x-4 md:gap-x-8 items-center justify-between px-4"
        >
          <div
            class="logo flex items-center gap-x-4"
          >
            <a
              aria-label="Visit ExploreIt Homepage"
              class="block"
              href="/"
            >
              <span
                class="font-bold text-black text-2xl"
              >
                ReactPress
              </span>
            </a>
          </div>
          <div
            class="grow"
          >
            <div
              class="text-right"
            >
              <button
                aria-label="Open search"
                class="sm:hidden text-primary text-2xl p-1"
                type="button"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </button>
            </div>
            <form
              class="relative searchForm  "
            >
              <div
                class="relative"
              >
                <input
                  aria-label="Search"
                  class="bg-white p-2 rounded-sm border sm:border-transparent w-full focus:outline-0 focus:border-primary text-gray-900"
                  placeholder="Enter WordPress powered website URL"
                  type="search"
                  value=""
                />
                <button
                  class="bg-transparent border-0 text-2xl px-4 cursor-pointer flex items-center absolute top-0 right-0 bottom-0 z-20"
                  type="search"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </button>
              </div>
            </form>
          </div>
          <a
            class="px-3 py-1 block font-bold rounded-sm bg-primary text-white"
            href="/"
          >
            <span
              class="flex items-center gap-x-1"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 496 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                />
              </svg>
              Github
            </span>
          </a>
        </div>
      </div>
    </header>
    <div
      class="container py-4"
    >
      <main
        class=""
      >
        <section>
          <div
            class="container px-4 flex items-center justify-center welcomeMessage"
          >
            <div
              class="sm:w-2/3 sm:max-w-[600px] text-center"
            >
              <h1
                class="font-bold text-3xl mb-4"
              >
                Welcome to our ReactPress!
              </h1>
              <p
                class="mb-2"
              >
                Simply enter the URL of any WordPress-powered website and retrieve the latest posts instantly. Whether you're exploring content or testing API integration, this tool makes it quick and effortless. 
              </p>
              <p
                class="italic mb-3"
              >
                Note: This tool only fetches posts of the "post" post type. Some websites may restrict API access, which could prevent fetching.
              </p>
              <p
                class="font-semibold"
              >
                Start your search now and unlock fresh insights! ðŸš€
              </p>
            </div>
          </div>
        </section>
      </main>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App.test.jsx Should have the 'Vite + React' text visible in the User Agent 1`] = `
<DocumentFragment>
  <div>
    <header
      class="bg-orange-100 py-2"
    >
      <div
        class="container"
      >
        <div
          class="flex gap-x-4 md:gap-x-8 items-center justify-between px-4"
        >
          <div
            class="logo flex items-center gap-x-4"
          >
            <a
              aria-label="Visit ExploreIt Homepage"
              class="block"
              href="/"
            >
              <span
                class="font-bold text-black text-2xl"
              >
                ReactPress
              </span>
            </a>
          </div>
          <div
            class="grow"
          >
            <div
              class="text-right"
            >
              <button
                aria-label="Open search"
                class="sm:hidden text-primary text-2xl p-1"
                type="button"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </button>
            </div>
            <form
              class="relative searchForm  "
            >
              <div
                class="relative"
              >
                <input
                  aria-label="Search"
                  class="bg-white p-2 rounded-sm border sm:border-transparent w-full focus:outline-0 focus:border-primary text-gray-900"
                  placeholder="Enter WordPress powered website URL"
                  type="search"
                  value=""
                />
                <button
                  class="bg-transparent border-0 text-2xl px-4 cursor-pointer flex items-center absolute top-0 right-0 bottom-0 z-20"
                  type="search"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </button>
              </div>
            </form>
          </div>
          <a
            class="px-3 py-1 block font-bold rounded-sm bg-primary text-white"
            href="/"
          >
            <span
              class="flex items-center gap-x-1"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 496 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                />
              </svg>
              Github
            </span>
          </a>
        </div>
      </div>
    </header>
    <div
      class="container py-4"
    >
      <main
        class=""
      >
        <section>
          <div
            class="container px-4 flex items-center justify-center welcomeMessage"
          >
            <div
              class="sm:w-2/3 sm:max-w-[600px] text-center"
            >
              <h1
                class="font-bold text-3xl mb-4"
              >
                Welcome to our ReactPress!
              </h1>
              <p
                class="mb-2"
              >
                Simply enter the URL of any WordPress-powered website and retrieve the latest posts instantly. Whether you're exploring content or testing API integration, this tool makes it quick and effortless. 
              </p>
              <p
                class="italic mb-3"
              >
                Note: This tool only fetches posts of the "post" post type. Some websites may restrict API access, which could prevent fetching.
              </p>
              <p
                class="font-semibold"
              >
                Start your search now and unlock fresh insights! ðŸš€
              </p>
            </div>
          </div>
        </section>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
